bin/main.rb

```diff
@@ -47,30 +56,42 @@ def display_board
     puts " #{@board[6]} | #{@board[7]} | #{@board[8]} "
   end
 
+  def position
+    current = @game.current_player
+    if current == 'X'
+      print "\n #{@player1.name}, choose a position between 1-9: "
+    else
+      print "\n #{@player2.name}, choose a position between 1-9: "
+    end
+  end
+
     # Game begins, player 1 starts choosing between 1-9
   # if the input is different valid_move returns false and input is request again
   def turn
-    current = @game.current_player
-    puts "Player #{current}, choose a spot between 1-9"
+    position
     spot = gets.strip
     spot = @game.board_index(spot)
     if @game.valid_move?(spot)
       @game.set_input(spot, @game.current_player)
+      display_board
     else
-      puts 'This spot hasbeen taken or you inputed a wrong value'
+      puts 'This spot has been taken or you inputed a wrong value'
       puts 'Please enter a valid option between 1-9'
       display_board
       turn
     end
-    display_board
   end
 
   # Turn method is called for players to fill the board until @game.over? method is true
   # If total moves = 9 and no combinations detected, its a draw, if a combination is detected, @game.won is executed
   def play
     turn until @game.over?
       if @game.won?
-        puts "Congratulations #{@board[@game.won?.first]}! You Won"
+        if @board[@game.won?.first] == 'X'
+        puts "CONGRATULATIONS #{@player1.name}! You Won"
+        else
+          puts "CONGRATULATIONS #{@player2.name}! You Won"
+        end
```

Comment:
```@board``` is already an attribute of Game class, avoid using it outside again.
Create a method into Game class for that snippet 
```ruby
def winner
    if @board[@game.won?.first] == :X
        return :X
    else
        return :O
    end
end
```

and use it in UserInterface
```ruby
if @game.winner == :X
    # add something ...
else
    # add something ...
end
```